doc/users_guide/command_interface.md

```diff
@@ -0,0 +1,570 @@
+This section focuses on the most commonly used Geant4 commands relevant in the context of a ratpac-two simulation.
+
+
+### 2.1. Run Control
+
+These commands manage the initialization and execution of simulation runs and dictate the order of commands passed to ratpac-two in a macro.
+
+* `/run/initialize`: This is one of the most critical commands in any Geant4-based simulation. 
+      It must be issued before the first run can start. 
+      This command triggers the construction of the detector geometry, calculation of physics tables, and preparation of user actions. 
+      In ratpac-two, it ensures that all detector parameters from RATDB are processed and the simulation world is built.
+
+   The typical placement of this command is after setting up detector parameters (e.g., via `/rat/db/` commands) but before defining event processors or starting the event loop.
+
+   **Note:** The run manager must be initialized *after* the geometry has been fully configured and *before* issuing any `/run/beamOn` command.
+   Calling `/run/beamOn` prior to `/run/initialize` will cause Geant4 to halt with an error similar to `G4RunManager::BeamOn() called before initialization`.
+
+ * `/run/beamOn <numberOfEvents>`: This command starts a simulation run, processing the specified number of events.
+
+   Multiple `/run/beamOn` commands can appear in a single macro, for instance, to simulate different particle types or energies sequentially after reconfiguring the event generator.
+
+
+### 2.2. Verbosity Control
+
+Geant4 provides extensive control over the amount of information printed during a simulation.
+These commands are invaluable for debugging and understanding the simulation process.
+Verbosity is typically controlled by an integer level, where 0 means minimal output and higher values provide more detail.
+
+ * `/run/verbose <level>`: Controls the verbosity of the run manager (e.g., messages about run initialization and termination).
+ * `/event/verbose <level>`: Controls verbosity related to event processing (e.g., information at the beginning and end of each event).
+ * `/tracking/verbose <level>`: Provides detailed information about particle tracking, including step-by-step details of particle interactions and movement through the geometry.
+      This is particularly useful for debugging physics processes or identifying issues with the detector geometry.
+  
+By strategically increasing verbosity for specific components like tracking or particular physics processes, new users can gain a much clearer picture of what Geant4 (and by extension ratpac-two) is doing "under the hood" for each event. 
+Observing this textual output can be more illuminating for learning than just examining final results or visualizations, as it connects abstract concepts of particle interactions and geometry definitions to concrete simulation steps.
+
+
+### 2.3. Detector Visualization

```

Comment:
In this section, it would be useful to describe what would happen if you try to run the visualization code without the `--vis` flag (I think RAT may crash?).